service_storagefacility/app/repositories/storage/old_dao/event/EventDao.scala

```diff
@@ -579,6 +578,33 @@ class EventDao @Inject()(
   }
 
   /**
+   * Function used for migration of data from old event table(s) to new.
+   */
+  def getAllEvents[A <: TopLevelEvent, Res <: MusitEvent_Old](
+      mid: MuseumId,
+      eventType: A
+  )(success: EventDto => Res): Future[Seq[Res]] = {
+    val getIds = eventBaseTable.filter(_.eventTypeId === eventType.id).map(_.id).result
+
+    db.run(getIds).flatMap { ids =>
+      Future.traverse(ids)(eid => getEvent(mid, eid, Some(eventType.id))).map { res =>
+        res
+          .filter(_.isSuccess)
+          .map {
+            case MusitSuccess(ms) =>
+              ms.map(dto => success(dto))
+
+            case err: MusitError =>
+              logger.error(err.message)
+              throw new IllegalStateException("Impossible state") // scalastyle:ignore
+          }
+          .filter(_.isDefined)
+          .map(_.get)
```

Comment:
Yes...in any other scenario this would be a no-go. But here it's actually just convenience for the migration. After migration this entire file will be removed. So I wouldn't be too concerned about it.